app/static/src/app/components/GraphSettings.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+    <div>
+        <div class="log-scale-y-axis form-check form-check-inline">
+            <label class="form-check-label">Log scale y axis</label>
+            <input type="checkbox" class="form-check-input" style="vertical-align: bottom" v-model="logScaleYAxis" />
+        </div>
+        <div class="lock-y-axis form-check form-check-inline">
+            <label class="form-check-label">Lock y axis</label>
+            <input type="checkbox" class="form-check-input" style="vertical-align: bottom" v-model="lockYAxis" />
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed } from "vue";
+import { useStore } from "vuex";
+import { GraphsMutation } from "../store/graphs/mutations";
+
+export default defineComponent({
+    name: "GraphSettings",
+    props: {
+        graphIndex: {
+            type: Number,
+            required: false,
+            default: -1
```

Comment:
If you don't provide the prop it should be because fitPlot is true. We could explicitly make this nullable and make null the default, but that's going to clutter up the code with null checks. Or we could squash both props into one and assume fitPlot if index is -1. But I think it's better to be explicit and it's probably ok to assume polite usage - but I'll add an explanatory comment.

Reply:
Understood, I hadn't thought about the null checks.